### PR TITLE
Add Swank server for SLIME integration

### PR DESCRIPTION
**Pull request by jduhamel**
## Summary

- Add a Swank protocol server (`pkg/swank/`) enabling SLIP to work as a backend for Emacs SLIME, supporting eval, completion, compilation, inspector, macroexpand, trace, and xref RPCs
- Comprehensive test suite (`test/swank/`) with 95%+ coverage across RPC handlers, connection lifecycle, edge cases, and wire protocol
- Emacs Lisp test files (`elisp/`) for end-to-end validation against a running server

### CODE REVIEW

File: cmd/slip/main.go [hunk 1]
```diff
@@ -46,6 +49,7 @@ func init() {
 			args = append(args, slip.String(s))
 			return nil
 		})
+	flag.StringVar(&emacsMode, "emacs", "", "start Emacs SLIME integration server")
```
Comment (ohler55) [inline cmd/slip/main.go]: What are valid options for this? slime?

File: elisp/test-swank-ert.el [hunk 2]
```diff
@@ -0,0 +1,425 @@
+;;; test-swank-ert.el --- ERT tests for Swank servers -*- lexical-binding: t -*-
+
+;; Copyright (c) 2025, Peter Ohler, All rights reserved.
```
Comment (ohler55) [inline elisp/test-swank-ert.el:3]: You wrote this. How about your name for the copyright?

File: pkg/swank/connection.go [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+func NewConnection(id int64, conn net.Conn, server *Server) *Connection {
+	// Create a child scope for this connection
+	scope := server.baseScope.NewScope()
+
+	c := &Connection{
+		id:          id,
+		conn:        conn,
+		server:      server,
+		scope:       scope,
+		currentPkg:  slip.FindPackage("cl-user"),
```
Comment (ohler55) [inline pkg/swank/connection.go]: could user `slip.CLPkg`

File: pkg/swank/connection.go [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+	}
+
+	// Wire cooperative interrupt into scope
+	c.scope.InterruptCheck = func() {
+		select {
+		case <-c.interruptCh:
+			panic(&slip.Panic{Message: "Keyboard interrupt"})
+		default:
+		}
+	}
```
Comment (ohler55) [inline pkg/swank/connection.go:59]: How is this different than just `<-c.interuptch`? Is there something special about the switch with a default?

Comment (ohler55) [reply pkg/swank/connection.go:59]: Looked it up. Neat trick. I'll keep that in mind for the future.

File: pkg/swank/server.go [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+			Text: `__create-server__ starts a Swank server for SLIME integration.
+This is the standard SLIME entry point. Equivalent to __swank-server__.`,
+			Examples: []string{
+				`(create-server :port 4005) => #<swank-server :4005>`,
+			},
+		}, &Pkg)
+}
+
+// CreateServer represents the create-server function.
+type CreateServer struct{ slip.Function }
```
Comment (ohler55) [inline pkg/swank/server.go]: For consistency this should be on multiple lines.

**Comment by ohler55** [comment]
Very impressive. I'll have to get a walk through of swank/slime in use with emacs when you can spare a few minutes.

**Comment by ohler55** [comment]
I noticed the verbose option is locked into colors for a black background. How do they colors look on white or is some other option needed?

There are two test files in the swank pkg directory that use private functions and vars for testing. Was it not possible or too difficult to test outside the package?

File: pkg/swank/connection.go (hunk 3 above)
Comment (jduhamel) [reply pkg/swank/connection.go]: Shouldn't it be cl-user I thought CLpkg was the :common-lisp ?

File: elisp/test-swank-ert.el (hunk 2 above)
Comment (jduhamel) [reply elisp/test-swank-ert.el:3]: I just use yours since it's your repo.

File: cmd/slip/main.go (hunk 1 above)
Comment (jduhamel) [reply cmd/slip/main.go]: so that was really for testing honestly the options would be start slime on a port. not sure whether to leave there or not?

File: pkg/swank/connection.go (hunk 4 above)
Comment (jduhamel) [reply pkg/swank/connection.go:59]: yeah blocking vs non-blocking.

File: cmd/slip/main.go (hunk 1 above)
Comment (ohler55) [reply cmd/slip/main.go]: Lets discuss.

File: elisp/test-swank-ert.el (hunk 2 above)
Comment (ohler55) [reply elisp/test-swank-ert.el:3]: Not sure of the legal stuff but if feels wrong to not give you credit. I suppose the git details contain that though.

File: pkg/swank/connection.go (hunk 3 above)
Comment (ohler55) [reply pkg/swank/connection.go]: You are right, `slip.UserPkg` then

Comment (jduhamel) [reply pkg/swank/connection.go]: swapped slip.FindPackage("cl-user") for &slip.UserPkg in NewConnection. Also skips the map lookup on each connect.

File: pkg/swank/server.go (hunk 5 above)
Comment (jduhamel) [reply pkg/swank/server.go]: done

**Review by ohler55 — APPROVED** [review]
